src/main/java/com/mongodb/kafka/connect/util/TimeseriesValidation.java

```diff
@@ -0,0 +1,273 @@
+import com.mongodb.kafka.connect.sink.namespace.mapping.DefaultNamespaceMapper;
+import com.mongodb.kafka.connect.sink.namespace.mapping.NamespaceMapper;
+
+public final class TimeseriesValidation {
+
+  private static final String COLLSTATS = "collStats";
+  private static final String TIMESERIES = "timeseries";
+  private static final String TOPIC_OVERRIDE_PREFIX = "topic.override.";
+
+  public static void validateConfigAndCollection(
+      final MongoClient mongoClient, final MongoSinkTopicConfig topicConfig, final Config config) {
+
+    validateConfigAndTimeseriesSupport(mongoClient, topicConfig, config);
+    if (!topicConfig.isTimeseries()
+        || config.configValues().stream().anyMatch(cv -> !cv.errorMessages().isEmpty())) {
+      return;
+    }
+
+    NamespaceMapper namespaceMapper = topicConfig.getNamespaceMapper();
+    if (namespaceMapper instanceof DefaultNamespaceMapper) {
+      getConfigByName(config, TIMESERIES_TIMEFIELD_CONFIG)
+          .ifPresent(
+              configValue -> {
+                try {
+                  validateCollection(
+                      mongoClient, namespaceMapper.getNamespace(null, null), topicConfig);
+                } catch (Exception e) {
+                  configValue.addErrorMessage(e.getMessage());
+                }
+              });
+    }
+  }
+
+  public static void validTopicRegexConfigAndCollection(
+      final MongoClient mongoClient, final MongoSinkConfig sinkConfig, final Config config) {
+    assert sinkConfig.getTopicRegex().isPresent();
+    Pattern pattern = sinkConfig.getTopicRegex().get();
+
+    Set<String> topicsWithOverrides =
+        sinkConfig.getOriginals().keySet().stream()
+            .filter(k -> k.startsWith(TOPIC_OVERRIDE_PREFIX))
+            .map(k -> k.substring(TOPIC_OVERRIDE_PREFIX.length()))
+            .map(
+                k -> {
+                  int index = k.indexOf(".");
+                  return k.substring(0, index > 0 ? index : k.length());
+                })
+            .filter(k -> k.matches(pattern.pattern()))
+            .collect(Collectors.toSet());
+
+    if (!topicsWithOverrides.isEmpty()) {
+      topicsWithOverrides.forEach(
+          t ->
+              validateConfigAndCollection(
+                  mongoClient,
+                  new MongoSinkTopicConfig(t, sinkConfig.getOriginals(), false),
+                  config));
+    } else {
+      validateConfigAndCollection(
+          mongoClient,
+          new MongoSinkTopicConfig("__default", sinkConfig.getOriginals(), false),
+          config);
+    }
+  }
+
+  public static void validateCollection(
+      final MongoClient mongoClient,
+      final MongoNamespace namespace,
+      final MongoSinkTopicConfig config) {
+
+    try {
+
+      Document collStats = getCollStats(mongoClient, namespace);
+      if (collStats.containsKey(TIMESERIES)) {
+        return;
+      }
+      if (collStats.getInteger("nindexes") > 0) {
```

Comment:
What is achieved by the `nindexes` check? It seems excessive to me for the following reasons:

- [`collStats`](https://docs.mongodb.com/v5.0/reference/command/collStats/) documentation states: _"If the collection does not exist, collStats returns an error message"_, which I expect is represented by an exception in the Java driver. Therefore, if we are executing the `collStats.getInteger("nindexes") > 0` check, we already know that the collection identified by `namespace` exists.
- The [Technical Design: Time-series Collection](https://docs.google.com/document/d/11GqNugOwdnwgU8LyOziLdrYhHTICX3UpcUFJVvU5Lw8/edit#heading=h.i4sji0le9arb) states _"A collStats command being run against a time-series collection contains the following additional fields ... `timeseries`"_. Taking into account 1, we know that if we are executing the `collStats.getInteger("nindexes") > 0` check, the collection not only exists but also is not a time-series collection.

Reply:
So this is a check to see if the collection actually exists.  Unfortunately the documentation doesn't reflect reality.

```
MongoDB Enterprise > show databases
admin   0.000GB
config  0.000GB
local   0.000GB
MongoDB Enterprise > use test
switched to db test
MongoDB Enterprise > var collStats = db.runCommand({collStats: "normal1"});
MongoDB Enterprise > collStats
{
	"ns" : "test.normal1",
	"size" : 0,
	"count" : 0,
	"storageSize" : 0,
	"totalSize" : 0,
	"nindexes" : 0,
	"totalIndexSize" : 0,
	"indexDetails" : {
		
	},
	"indexSizes" : {
		
	},
	"scaleFactor" : 1,
	"ok" : 1
}
```

There is no error on 4.2, 4.4 or the latest rc. So I can use the same command to determine if the collection exists.

Reply:
Looks like it errored in 3.6 and previous. As this code is already guarded against pre 5.0 so it won't error. 
Added DOCSP-17199 for the docs to be updated.

Reply:
@rozza, thank you for the detailed explanation and for filing DOCSP-17199!